components/supervisor/pkg/ports/ports_test.go

```diff
@@ -443,6 +443,56 @@ func TestPortsUpdateState(t *testing.T) {
 				{{LocalPort: 5900, Served: true, Exposed: &api.ExposedPortInfo{Visibility: api.PortVisibility_private, OnExposed: api.OnPortExposedAction_notify_private, Url: "foobar"}}},
 			},
 		},
+		{
+			Desc: "port status has description set as soon as the port gets exposed, if there was a description configured",
+			Changes: []Change{
+				{
+					Config: &ConfigChange{workspace: []*gitpod.PortConfig{
+						{Port: 8080, Visibility: "private", Description: "Development server"},
+					}},
+				},
+				{
+					Served: []ServedPort{{"00000000", 8080, false}},
+				},
+				{
+					Exposed: []ExposedPort{{LocalPort: 8080, Public: false, URL: "foobar"}},
+				},
+			},
+			ExpectedExposure: []ExposedPort{
+				{LocalPort: 8080},
+			},
+			ExpectedUpdates: UpdateExpectation{
+				{},
+				{{LocalPort: 8080, Description: ""}},
+				{{LocalPort: 8080, Description: "", Served: true}},
+				{{LocalPort: 8080, Description: "Development server", Served: true, Exposed: &api.ExposedPortInfo{Visibility: api.PortVisibility_private, OnExposed: api.OnPortExposedAction_notify, Url: "foobar"}}},
```

Comment:
This was unexpected, I thought the port status would have the description set since the beginning, but instead, the description was set only after the port got exposed.
Not sure if it's an error or not.

Reply:
yes, from user perspective description should be available as soon as we read configuration, serving and exposing not necessary.

Have you tried to debug to find out why it happens?

Reply:
Thanks for confirming! I'll debug this tonight.

Reply:
Fixed by https://github.com/gitpod-io/gitpod/pull/7012/commits/43b26f216fd010d0f60ec380d09f53811d976e48.
The problem was the port description was set on PortStatus only when the exposed status was reached.
Now it's set as soon the configuration is available.